plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -350,7 +368,15 @@ func (c *nodePlugin) admitNode(nodeName string, a admission.Attributes) error {
 		if len(requestedName) == 0 {
 			requestedName = node.Name
 		}
+		// Taint node with NotReady taint at creation if TaintNodesByCondition is
+		// enabled. This is needed to make sure that nodes are added to the cluster
+		// with the taint. Otherwise, a new node may receive the taint with some delay
```

Comment:
nit: s/the taint/the NotReadytaint.

the comment could be further expanded to something like `the new node may receive taint from nodecontroller with some delay..`